src/algebra/ring/ulift.lean

```diff
@@ -82,4 +82,13 @@ by refine_struct { zero := (0 : ulift α), one := 1, add := (+), mul := (*), sub
   zsmul := sub_neg_monoid.zsmul };
 tactic.pi_instance_derive_field
 
+instance field [field α] : field (ulift α) :=
+begin refine_struct { zero := (0 : ulift α), one := 1, add := (+), mul := (*), sub := has_sub.sub,
```

Comment:
Ideally we'd be able to use `function.injective.field` here, but we need #12126 for that to work properly

Reply:
This file is done not in the style of `group/ulift`, so is it worth refactoring fully for that when #12126 merges? Or merge this first and fix there? I don't mind, just don't want you to have to do too much extra work

Reply:
We could also just use `equiv.ulift.field` from the `data.equiv.transfer_instance` field.

Reply:
We don't want to do that, those instances don't set the `nsmul` and `zsmul` fields correctly.

Reply:
I'd like to do as little refactoring as possible in that PR, just to stop the scope exploding too much. I can't really see an easy way to avoid doing everything at once - fixing the instances downstream is a pain without changing these lemmas, while changing these lemmas breaks all the instances downstream.

Reply:
I see. I'll file this away as a followup to that fix, and if I can help with it let me know! I'll let you have the final say on what to tell `bors` now :)